src/grappler/ngraph_optimizer.cc

```diff
@@ -166,8 +191,32 @@ Status NgraphOptimizer::Optimize(tensorflow::grappler::Cluster* cluster,
     DumpGraphs(graph, idx, "unmarked", "Unmarked Graph");
   }
 
+  // Get backend + its configurations, to be attached to the nodes
+  // Precedence Order: RewriteConfig> Env Variable > BackendManager
+  string backend_name = BackendManager::GetCurrentlySetBackendName();
+  if (!config_backend_name.empty()) {
+    backend_name = config_backend_name;
+  } else {
+    const char* ng_backend_env_value = std::getenv("NGRAPH_TF_BACKEND");
+    if (ng_backend_env_value != nullptr) {
```

Comment:
So - is the backend selection logic common? may be we can refactor this into a function and use for both tracks?

Reply:
We will change the BackendManager::GetCurrentlySetBackendName() function to return the backend set by the environment variable in a future PR.